controllers/storagecluster/ocs_operator_config.go

```diff
@@ -88,3 +93,11 @@ func (r *StorageClusterReconciler) getClusterID() string {
 	}
 	return fmt.Sprint(clusterVersion.Spec.ClusterID)
 }
+
+// getCephFSKernelMountOptions returns the kernel mount options for cephfs
+func getCephFSKernelMountOptions(sc *ocsv1.StorageCluster) string {
+	if sc.Spec.Network != nil && sc.Spec.Network.Connections != nil && sc.Spec.Network.Connections.Encryption != nil && sc.Spec.Network.Connections.Encryption.Enabled {
+		return "ms_mode=secure"
+	}
+	return "ms_mode=prefer-crc"
```

Comment:
Changes LGTM, I hope you have tested how this behaves when v1 ports are used in upgraded clusters.

Reply:
Haven't been able to test the case with upgrades with Mon failover

Reply:
Let me do it